flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/ScalarOperatorsTest.scala

```diff
@@ -94,10 +94,33 @@ class ScalarOperatorsTest extends ScalarOperatorsTestBase {
       "x'68656C6C6F20636F6465' > x'68656C6C6F2063617374'",
       "true")
 
+    // between array and array
```

Comment:
Thanks @wangxlong. I will do a full review soon. Could you use the new test base? 
`org.apache.flink.table.planner.functions.CastFunctionITCase`

It was recently added for casting rows to rows. It will be further extended as part of FLINK-17321.

Reply:
Thanks @twalthr. Yes, I will update it soon.

Reply:
It is ready to be reviewed. Thanks for having a look in your free time. @twalthr